### PR TITLE
RFC: Only use camptocamp/systemd if available

### PR DESCRIPTION
**Pull request by alexjfisher**
Untested POC for discussion.

### CODE REVIEW

**Comment by bastelfreak** [comment]
Personally I'm not a big fan of this. In my opinion this add too much boilerplate code. I would like to get the feedback from the other @voxpupuli/collaborators

File: manifests/config.pp [hunk 1]
```diff
@@ -6,6 +6,7 @@
   $scrape_configs,
   $remote_read_configs,
   $remote_write_configs,
+  Boolean $use_camptocamp_systemd = $::prometheus::params::use_camptocamp_systemd,
```
Comment (alexjfisher) [inline manifests/config.pp:9]: BTW, I don't think *subclasses* inheriting params.pp is a great pattern.  I would normally want to be able to override the parameter in the base class (in one place) and have that take effect in all the subclasses.  But this is consistent with the other parameters and fits the way the testing is currently setup.  (There's no `let(:pre_condition) { include prometheus }` stuff in there).  Also would anybody ever want/need to override the autodetection code in params.pp??

File: manifests/config.pp [hunk 2]
```diff
@@ -67,9 +68,25 @@
...
+          include 'systemd'
+          ::systemd::unit_file {'prometheus.service':
+            content => $systemd_content,
+          }
+        } else {
+          file { '/etc/systemd/system/prometheus.service':
+            mode    => '0644',
+            owner   => 'root',
+            group   => 'root',
+            notify  => $systemd_notify,
```
Comment (alexjfisher) [inline manifests/config.pp:82]: `$systemd_notify` is defined further up.  It was never taken out of the code in https://github.com/voxpupuli/puppet-prometheus/pull/90

File: metadata.json [hunk 3]
```diff
@@ -15,10 +15,6 @@
     {
       "name": "puppetlabs/stdlib",
       "version_requirement": ">= 4.20.0 < 5.0.0"
-    },
-    {
-      "name": "camptocamp/systemd",
```
Comment (alexjfisher) [inline metadata.json:20]: It becomes a 'soft' dependency.  If somebody wants to take this PR on (maybe add a test or two for `$use_camptocamp_systemd => false`), then the README should probably get a sentence explaining how this works.

**Comment by alexjfisher** [comment]
Closing for now so that we can keep discussion over in https://github.com/voxpupuli/puppet-prometheus/pull/90. Can reopen later if people like the idea.  Or perhaps someone will cherry-pick my commit and add some tests, README update etc and open a new PR.

File: manifests/params.pp [hunk 4]
```diff
@@ -271,4 +271,16 @@
   if $init_style == undef {
     fail('Unsupported OS')
   }
+
+  # Autodetect whether we should use camptocamp/systemd
```
Comment (juniorsysadmin) [inline manifests/params.pp:275]: I don't like this because I don't think it will work as you expect

File: metadata.json (hunk 3 above)
Comment (juniorsysadmin) [reply metadata.json:20]: I like removal of hard dependencies

File: manifests/params.pp (hunk 4 above)
Comment (alexjfisher) [reply manifests/params.pp:275]: How come?
It worked last time I used it.
https://github.com/cpitman/puppet-database_schema/pull/4

File: manifests/params.pp [hunk 5]
```diff
@@ -271,4 +271,16 @@
...
+  $metadata = load_module_metadata('systemd', true)
+  if empty($metadata) {
```
Comment (alexjfisher) [inline manifests/params.pp:277]: `load_module_metadata` didn't quite do what I expected here.  It only returns empty metadata if the metadata didn't exist.  If the module `systemd` didn't exist at all, `load_module_metadata` throws a parse error. :(

**Pull request closed without merging**